osclib/conf.py

```diff
@@ -244,18 +203,32 @@ def read_section(self, section, defaults):
         else:
             return defaults
 
+    def _migrate_staging_config(self, api):
+        # first try staging project's dashboard. Can't rely on cstaging as it's
+        # defined by config
+        config = api.load_file_content(self.project + ':Staging', 'dashboard', 'config')
+        if not config:
+            config = api.load_file_content(self.project, 'dashboard', 'config')
+        if not config:
+            return None
+
+        print("Found config in staging dashboard - migrate now [y/n] (y)? ", end='')
+        response = raw_input().lower()
+        if response != '' and response != 'y':
+            return config
+
+        api.attribute_value_save('Config', config)
+
     def apply_remote(self, api):
         """Fetch remote config and re-process (defaults, remote, .oscrc)."""
-        if not conf.config[self.project].get('remote-config', True):
-            return
 
-        config = api.dashboard_content_load('config')
+        config = api.attribute_value_load('Config')
+        if not config:
+            # try the old way
+            config = self._migrate_staging_config(api)
         if config:
             cp = ConfigParser()
             config = '[remote]\n' + config
             cp.readfp(io.BytesIO(config))
             self.remote_values = dict(cp.items('remote'))
             self.populate_conf()
-        elif config is None:
-            # Write empty config to allow for caching.
-            api.dashboard_content_save('config', '')
```

Comment:
To be clear, this will mean a remote request looking for config every time for projects that do not have one since 404's are not cached. This was avoided with remote-config option locally. Obviously, if opening this up to any project it is likely not desired to write empty config.